app/src/androidTest/java/com/github/sdp_begreen/begreen/fragments/CameraContainerTest.kt

```diff
@@ -0,0 +1,150 @@
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import androidx.test.rule.GrantPermissionRule
+import com.github.sdp_begreen.begreen.R
+import com.github.sdp_begreen.begreen.activities.MainActivity
+import com.github.sdp_begreen.begreen.espressoUtils.BaseRobot
+import com.github.sdp_begreen.begreen.fragments.CameraContainer
+import com.github.sdp_begreen.begreen.firebase.Auth
+import com.github.sdp_begreen.begreen.firebase.DB
+import com.github.sdp_begreen.begreen.models.User
+import com.github.sdp_begreen.begreen.rules.KoinTestRule
+import com.github.sdp_begreen.begreen.viewModels.ConnectedUserViewModel
+import com.google.android.material.internal.ContextUtils.getActivity
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.test.runTest
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.koin.dsl.module
+import org.mockito.Mockito
+
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class CameraContainerTest {
+    private lateinit var fragmentScenario: FragmentScenario<CameraContainer>
+    private lateinit var fragmentManager: FragmentManager
+
+    //Companion object to mock the DB and Auth
+    companion object {
+        val user = User("test", 2, "test", 5, null, "test", "test", "test", 15)
+        private val db: DB = Mockito.mock(DB::class.java)
+        private val auth: Auth = Mockito.mock(Auth::class.java)
+        val users = listOf<User>(
+            User("1", 123, "Alice"),
+            User("2", 0, "Bob Zeu bricoleur"),
+            User("3", 14, "Charlie Chaplin"),
+            User("4", 23, "David Pujadas"),
+            User("5", 10492, "Euler"),
+            User("6", 1234, "Alain Berset"),
+            User("7", 1235, "Mister Alix")
+        )
+        // Setup the mock
+        @OptIn(ExperimentalCoroutinesApi::class)
+        @BeforeClass
+        @JvmStatic
+        fun setUp() {
+            runTest {
+                // setup basic get user and getProfilePicture use in multiple tests
+                Mockito.`when`(db.getUser(user.id))
+                    .thenReturn(user)
+                // add a small delay, just to be sure that it is triggered after initialization
+                // and arrive second, after the initial null value
+                // user between tests, by simply pushing a new userId
+                Mockito.`when`(auth.getFlowUserIds())
+                    .thenReturn(MutableStateFlow(user.id))
+                Mockito.`when`(auth.getConnectedUserId())
+                    .thenReturn(user.id)
+                Mockito.`when`(db.getAllUsers()).thenReturn(users)
+            }
+        }
+    }
+
+    //Setup the activity rule
+    @get:Rule
+    val activityRule = ActivityScenarioRule(MainActivity::class.java)
```

Comment:
Why do you launch the `MainActivity` here ?

Reply:
It's a vestige of me trying to give mainFragmentContainer in the fragmentManager to run tests on isolation (didn't manage to)